homeassistant/components/opensprinkler/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Config flow for Opensprinkler integration."""
+import logging
+
+from pyopensprinkler import (
+    OpenSprinkler,
+    OpensprinklerAuthError,
+    OpensprinklerConnectionError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_PORT
+
+from .const import (  # pylint: disable=unused-import
+    CONF_RUN_SECONDS,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_RUN_SECONDS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Optional(CONF_RUN_SECONDS, default=DEFAULT_RUN_SECONDS): int,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Opensprinkler."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                password = user_input[CONF_PASSWORD]
+                host = (
+                    f"{user_input[CONF_HOST]}:{user_input.get(CONF_PORT, DEFAULT_PORT)}"
+                )
+                await self.hass.async_add_executor_job(OpenSprinkler, host, password)
+
+                await self.async_set_unique_id(host)
```

Comment:
host is not a unique ID, it can change when the device gets a new IP assigned.

Reply:
The only other option is the name which would probably be worse. I tried looking through the API but it doesn't have anything to uniquely identify the device.

Do you have any suggestions?

Reply:
I was planning on adding an input for mac address for this. As you said unfortunately it’s not available via the API currently but may be in the future.